src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -788,7 +877,302 @@ namespace OpenMS
+    }
+    // search params:
+    for (SearchParamRef other_ref = other.getDBSearchParams().begin();
+         other_ref != other.getDBSearchParams().end(); ++other_ref)
+    {
+      trans.search_param_refs[other_ref] = registerDBSearchParam(*other_ref);
+    }
+    // processing steps:
+    for (ProcessingStepRef other_ref = other.getProcessingSteps().begin();
+         other_ref != other.getProcessingSteps().end(); ++other_ref)
+    {
+      // update internal references:
+      ProcessingStep copy = *other_ref;
+      copy.software_ref = trans.processing_software_refs[copy.software_ref];
+      for (InputFileRef& file_ref : copy.input_file_refs)
+      {
+        file_ref = trans.input_file_refs[file_ref];
+      }
+      trans.processing_step_refs[other_ref] = registerProcessingStep(copy);
+    }
+    // search steps:
+    for (const auto& pair : other.getDBSearchSteps())
+    {
+      ProcessingStepRef step_ref = trans.processing_step_refs[pair.first];
+      SearchParamRef param_ref = trans.search_param_refs[pair.second];
+      db_search_steps_[step_ref] = param_ref;
+    }
+    // observations:
+    for (ObservationRef other_ref = other.getObservations().begin();
+         other_ref != other.getObservations().end(); ++other_ref)
+    {
+      // update internal references:
+      Observation copy = *other_ref;
+      copy.input_file = trans.input_file_refs[copy.input_file];
+      trans.observation_refs[other_ref] = registerObservation(copy);
+    }
+    // parent sequences:
+    for (ParentSequenceRef other_ref = other.getParentSequences().begin();
+         other_ref != other.getParentSequences().end(); ++other_ref)
+    {
+      // don't copy processing steps and scores yet:
+      ParentSequence copy(other_ref->accession, other_ref->molecule_type,
+                          other_ref->sequence, other_ref->description,
+                          other_ref->coverage, other_ref->is_decoy);
+      // now copy precessing steps and scores while updating references:
+      mergeScoredProcessingResults_(copy, *other_ref, trans);
+      trans.parent_sequence_refs[other_ref] = registerParentSequence(copy);
+    }
+    // identified peptides:
+    for (IdentifiedPeptideRef other_ref = other.getIdentifiedPeptides().begin();
+         other_ref != other.getIdentifiedPeptides().end(); ++other_ref)
+    {
+      // don't copy parent matches, steps/scores yet:
+      IdentifiedPeptide copy(other_ref->sequence, ParentMatches());
+      // now copy steps/scores and parent matches while updating references:
+      mergeScoredProcessingResults_(copy, *other_ref, trans);
+      for (const auto& pair : other_ref->parent_matches)
+      {
+        ParentSequenceRef parent_ref = trans.parent_sequence_refs[pair.first];
+        copy.parent_matches[parent_ref] = pair.second;
+      }
+      // @TODO: with C++17, 'map::extract' offers a better way to update keys
+      trans.identified_peptide_refs[other_ref] = registerIdentifiedPeptide(copy);
```

Comment:
we ave C++17 now. how would this look like?

Reply:
You could copy the whole `parent_matches` map of the peptide, then change the keys (i.e. references to parent sequences) using this approach:
https://en.cppreference.com/w/cpp/container/map/extract#Notes
But actually that doesn't seem easier or more efficient in the end.

Reply:
I'm removing the comment.